dotnet/src/Backends.HuggingFace/HuggingFaceBackend.cs

```diff
@@ -0,0 +1,176 @@
+    public HuggingFaceBackend(string baseUri, string model, HttpClientHandler httpClientHandler)
+        : this(new Uri(baseUri), model, httpClientHandler)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HuggingFaceBackend"/> class.
+    /// Using default <see cref="HttpClientHandler"/> implementation.
+    /// </summary>
+    /// <param name="baseUri">Base URI to call for backend operations.</param>
+    /// <param name="model">Model to use for backend operations.</param>
+    public HuggingFaceBackend(Uri baseUri, string model)
+        : this(baseUri, model, new HttpClientHandler())
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HuggingFaceBackend"/> class.
+    /// Using default <see cref="HttpClientHandler"/> implementation.
+    /// </summary>
+    /// <param name="baseUri">Base URI to call for backend operations in <see cref="string"/> format.</param>
+    /// <param name="model">Model to use for backend operations.</param>
+    public HuggingFaceBackend(string baseUri, string model)
+        : this(new Uri(baseUri), model)
+    {
+    }
+
+    /// <inheritdoc/>
+    public async Task<string> CompleteAsync(string text, CompleteRequestSettings requestSettings)
+    {
+        return await this.ExecuteCompleteRequestAsync(text);
+    }
+
+    /// <inheritdoc/>
+    public async Task<IList<Embedding<float>>> GenerateEmbeddingsAsync(IList<string> data)
+    {
+        return await this.ExecuteEmbeddingRequestAsync(data);
+    }
+
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        this._httpClient.Dispose();
+        this._httpClientHandler.Dispose();
+    }
+
+    #region private ================================================================================
+
+    /// <summary>
+    /// Performs HTTP request to given base URI for completion.
+    /// </summary>
+    /// <param name="text">Text to complete.</param>
+    /// <returns>Completed text.</returns>
+    /// <exception cref="AIException">Exception when backend didn't respond with completed text.</exception>
+    private async Task<string> ExecuteCompleteRequestAsync(string text)
+    {
+        try
+        {
+            var completionRequest = new CompletionRequest
+            {
+                Prompt = text,
+                Model = _model
+            };
+
+            using var httpRequestMessage = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Post,
+                RequestUri = new Uri(CompletionEndpoint, UriKind.Relative),
+                Content = new StringContent(JsonSerializer.Serialize(completionRequest)),
+            };
+
+            var response = await this._httpClient.SendAsync(httpRequestMessage).ConfigureAwait(false);
+            var body = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+
+            var completionResponse = JsonSerializer.Deserialize<CompletionResponse>(body);
```

Comment:
Need to handle serialization issues and have a very clear error message for problem determination

Reply:
Thanks, added serialization and HTTP response handling.